backend/app/api/routes/campaigns.py

```
@@ -0,0 +1,345 @@
+    try:
+        # Verify campaign exists and belongs to this brand
+        existing = supabase.table("campaigns").select("id").eq("id", campaign_id).eq("brand_id", brand_id).single().execute()
+
+        if not existing.data:
+            raise HTTPException(status_code=404, detail="Campaign not found")
+
+        # Prepare update data (only include non-None fields)
+        update_data = {k: v for k, v in campaign.dict().items() if v is not None}
+
+        if not update_data:
+            raise HTTPException(status_code=400, detail="No fields to update")
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)